### PR TITLE
Adding pipeline for DW testing.

### PR DESCRIPTION
**Pull request by rohkhann**
## Why make this change?
This change adds a pipeline to for integration testing of data warehouse changes.

## What is this change?
1. Added a testcategory for data warehouse. 
2. Added a pipeline for data warehouse query execution. Currently this will execute all queries against an mssql db. (both DW and mssql use t-sql underneath but there are differences and ideally we would want to change this to execute against a DW).

Issue to track: https://github.com/Azure/data-api-builder/issues/1878

### CODE REVIEW

File: .pipelines/dwsql-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      overWrite: true
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish received files as Artifacts'
+    name: 'verifypublish'
+    condition: eq(variables['publishverify'], 'Yes')
+    inputs:
+      PathtoPublish: '$(Build.ArtifactStagingDirectory)\Verify'
+      ArtifactName: 'Verify'
+      publishLocation: 'Container'
```
Comment (Aniruddh25) [inline .pipelines/dwsql-pipelines.yml:218]:
```suggestion
      publishLocation: 'Container'
      
```

**Review by Aniruddh25 — APPROVED** [review]
LGTM

File: config-generators/dwsql-commands.txt
Comment (seantleonard) [inline config-generators/dwsql-commands.txt:1]: does DwSql need the same config file as all the other tests?

File: src/Service.Tests/TestCategory.cs [hunk 2]
```diff
@@ -9,5 +9,6 @@ public class TestCategory
         public const string MSSQL = "MsSql";
         public const string POSTGRESQL = "PostgreSql";
         public const string MYSQL = "MySql";
+        public const string DWSQL = "DWSql";
```
[the remark this answers is not in the document]
Comment (rohkhann) [reply src/Service.Tests/TestCategory.cs]: Fixed!

File: config-generators/dwsql-commands.txt
Comment (rohkhann) [reply config-generators/dwsql-commands.txt:1]: We would eventually want to support stored procedures, mutations as well. Similar auth tests will also be there for DW. Will modify as necessary as more and more tests come in.

File: .pipelines/dwsql-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,218 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# DWSql Integration Testing Pipeline config is split into two jobs:
+# 1) LinuxTests -> Run SQL Server 2019 in Linux Docker Image
+# 2) WindowsTests -> Run LocalDB preinstalled on
```
[the remark this answers is not in the document]
Comment (rohkhann) [reply .pipelines/dwsql-pipelines.yml]: fixed.

File: .pipelines/dwsql-pipelines.yml [hunk 4]
```diff
@@ -215,4 +215,5 @@ jobs:
     inputs:
       PathtoPublish: '$(Build.ArtifactStagingDirectory)\Verify'
       ArtifactName: 'Verify'
-      publishLocation: 'Container'
\ No newline at end of file
+      publishLocation: 'Container'
+
```
Comment (Aniruddh25) [inline .pipelines/dwsql-pipelines.yml:219]:
```suggestion
      
 
```

nit: new line, not sure why but git UI complains last line in the file is missing

**Review by Aniruddh25 — APPROVED** [review]

File: .pipelines/dwsql-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish received files as Artifacts'
+    name: 'verifypublish'
+    condition: eq(variables['publishverify'], 'Yes')
+    inputs:
+      PathtoPublish: '$(Build.ArtifactStagingDirectory)\Verify'
+      ArtifactName: 'Verify'
+      publishLocation: 'Container'
+      
+
```
Comment (Aniruddh25) [inline .pipelines/dwsql-pipelines.yml:220]: For some reason, this file still shows that its missing a new line at the end of file:

image

**Review by seantleonard — APPROVED** [review]
thanks for addressing feedback!

File: .pipelines/dwsql-pipelines.yml (hunk 5 above)
Comment (rohkhann) [reply .pipelines/dwsql-pipelines.yml]: Added new line in pr again. Going to resolve this for now to unblock. Will take a look at it in the query building pr where ill actually use this pipeline.

**Pull request merged**